scipy/optimize/optimize.py

```diff
@@ -2048,16 +2051,16 @@ def golden(func, args=(), brack=None, tol=_epsilon, full_output=0):
     interval.
 
     """
-    options = {'xtol': tol}
+    options = {'xtol': tol, 'maxiter': maxiter}
     res = _minimize_scalar_golden(func, brack, args, **options)
     if full_output:
-        return res['x'], res['fun'], res['nfev']
+        return res['x'], res['fun'], res['nit'], res['nfev']
```

Comment:
This also breaks bw compat. I'd say we should only update the `minimize_scalar` return value (adding an attribute preserves bw compat) and keep `golden` return value as is.